llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -3497,6 +3695,8 @@ class VPlan {
   /// live-outs are fixed via VPLiveOut::fixPhi.
   MapVector<PHINode *, VPLiveOut *> LiveOuts;
 
+  MapVector<PHINode *, VPCSAState *> CSAStates;
```

Comment:
Why does this need to be part of VPlan? Is it not possible to encode all required info in recipes?

Reply:
removed